stage3/react/modules/module03/README.md

```diff
@@ -0,0 +1,57 @@
+# Неделя 3
+
+## React. API
+
+
+
+
+!! Задание выполняется в личном приватном репозитории !!
+Нужно выполнять задание на базе предыдущего
+
+
+
+Что должно быть сделано:
+
+
+
+1) Создать отдельную ветку для этого задания
+2) Выбрать апи
+Предлагается использовать следующие API(но можно использовать любое другое, которое вам нравится, если оно поддерживает поиск, пагинацию и сортировку): 
+      - https://www.flickr.com/services/api/flickr.photos.search.html
+      - https://newsapi.org/docs/endpoints/everything   
+      - https://the-one-api.dev/documentation
+**Обязательно, чтобы апи поддерживало поиск, пагинацию и сортировку**
+2) Переписать страницу для поиска, разбить ее на 2 логических части:
+    - **строка поиска.** Вводим текст -> нажимаем Enter -> отправляем запрос к API с введенным параметром -> список результатов поиска обновляется
+    - **список результатов поиска**: отображаем данные которые вернул API запрос при помощи Cards. 
+    Кастомер решил что требования поменялись.
+    Каждый элемент списка теперь должен отображать минимальную информацию. При клике на элемент - отображаем всю доступную информацию об этом элементе в модальном окне. Закрываться модальное окно должно по клику на крестик в верхнем правом углу или по нажатию на странизу за пределами модального окна. При открытии модального окна на страницу должен быть наложен overlay.
+
+3) Сделать индикатор загрузки (компонент с анимацией или хотя бы строка - "Загружаем"). Расположение индикатора на свое чувство прекрасного.
+
+Все логические части должны быть вынесены в компоненты.
+**User friendly интерфейс с индекатором загрузки и сообщениями на случай если данные не найдены или что-то сломалось приветствуются.**
+5) Добавить тесты, апи коллы в тестах замокать
+
+### Оценка
+
+
+
+
+Задание будет оцениваться ментором в марте. Для удобства проверки создайте Pull Request(**МЕРЖИТЬ НЕ НАДО**).
+
+
+
+
+1) Тесты - **3 балла**
+2) App + API - **12 баллов**
+
+
+
+
+Если хоть один из пунктов не выполнен, то задание оценивается в **0 баллов**.
+
+### Теория
+https://www.youtube.com/watch?v=qdCHEUaFhBk&list=PL4cUxeGkcC9gZD-Tvwfod2gaISzfRiP9d&index=17
```

Comment:
React API: https://www.youtube.com/watch?v=W9czPM2N65Y&t=3495s
Fetching Data with Fetch API: https://www.youtube.com/watch?v=r40gtnaTe9s
Fetch Data from an API in React: https://www.youtube.com/watch?v=T3Px88x_PsA